### ISSUE TITLE
Change dcterms creator/contributor usage

### ISSUE THREAD

**Issue by zednis** [USGCRP/gcis-ontology#2]
Proposal: [change dcterms:creator/contributor usage](https://github.com/USGCRP/gcis-ontology/wiki/Ontology-change-proposal---change-dcterms:creator-contributor-usage)
## Proposal

Change usage of dcterms:creator and dcterms:contributor properties to be as object-properties.
## Background

Currently the ontology is treating dcterms:creator and dcterms:contributor properties as datatype properties.

from [gcis.ttl](https://github.com/USGCRP/gcis-ontology/blob/master/gcis.ttl):

```
dcterms:creator "Xiaogang Ma <max7@rpi.edu>; Curt Tilmes <curt.tilmes@nasa.gov>; Peter Fox <foxp@rpi.edu>" ;
dcterms:contributor "Jin Guang Zheng <zhengj3@rpi.edu>; Justin Goldstein <jgoldstein@usgcrp.gov>; Linyun Fu <ful2@rpi.edu>; Brian Duggan <bduggan@ucar.edu>; Patrick West <westp@rpi.edu>; Steve Aulenbach <saulenbach@usgcrp.gov>; Chengcong Du <duc2@rpi.edu>; Jun Xu <xuj6@rpi.edu>; Anusha Akkiraju <akkira@rpi.edu>" ;
```

From my understanding these properties are object-properties (dc:creator/dc:contributor are the corresponding datatype properties)

example from http://dublincore.org/documents/dc-rdf/

```
<dcterms:creator>
   <foaf:Person rdf:about="http://example.org/person45">
    <rdf:value>John Smith</rdf:value>
    <foaf:phone rdf:resource="tel:+358-555-1234567"/>
   </foaf:Person>
  </dcterms:creator>
```
## Does this change affect the semantics of the ontology?

NO, but using these properties as datatype properties may affect the inferred profile of the ontology.  Using string values for these properties may put the ontology into OWL FULL, but I have yet to check if that is the current case.  
## Describe in detail the change being requested

Change usage of dcterms:creator and dcterms:contributor properties to be used as object-properties.  Create foaf:Person instances for each of the people referenced and add a name and email property for each.

The current modeling would become

```
dcterms:creator [ 
  a foaf:Person ;
  foaf:name "Xiaogang Ma" ;
  foaf:mbox <mailto:max7@rpi.edu>  
  ] .
```
## Describe the reason for the change
- dcterms:creator and dcterms:contributor should be used as object properties.  
- The ontology might be OWL FULL with the current usage.
## What are the anticipated impacts
- clients that expect to read and unpack the creator and contributor names from the string value will have to be updated
## What are the risks of not performing the change?
- The ontology profile may currently be OWL FULL
## What is the expected technical effort to support the change?
- low effort on ontology changes
- low effort to document the changes for users
- low effort to update clients that use these properties in their current usage
## Describe alternatives considered
- change usage of dcterms:creator and dcterms:contributor to dc:creator and dc:contributor, respectively.  The dublin core elements properties are intended to be used as datatype properties.
## Related change proposals

none
## Notes

none

**Comment by bduggan** [comment]
:thumbsup:

**Comment by zednis** [comment]
One thing I thought about after I wrote up the change request is ordering of the creator and contributor individuals.  Do we have a requirement to retain ordering?

Also, is there a preference to mint URIs or use blank nodes with the contributor and creator individuals?

**Comment by bduggan** [comment]
I would say there is no such requirement, and no preference; I would just go with whatever is common practice.

**Comment by zednis** [comment]
Ordering of authors is a common need, but I have seen it implemented in several ways.

We might not need it for the ontology creators and contributors, but I suspect we will want author ordering for the NCA report chapters - so I think it would be beneficial to have a discussion on how to best model it.

I checked the Linked Data Design Patterns site for a recommended practice.

Linked Data Design Patterns - Ordered List

http://patterns.dataincubator.org/book/ordered-list.html

Suggests using a rdf:List - which makes some sense but I don't generally see rdf:Lists used very much (I believe because they are poorly handled in SPARQL)

If we were to use an rdf:List the syntax would look like this in Turtle:

```
<http://data.globalchange.gov/gcis.owl> a owl:Ontology
  dcterms:creator ( gcis:xiaogang_ma gcis:curt_tilmes gcis:peter_fox ) .

gcis:xiaogang_ma
  a foaf:Person ;
  foaf:name "Xiaogang Ma" ;
  foaf:mbox <mailto:max7@rpi.edu> .

# ...
```

One thing I frequently see used in practice is use of a qualified relationship that includes an ordering relation.  This modeling can be used with the standard dcterms property.

http://patterns.dataincubator.org/book/qualified-relation.html
http://patterns.dataincubator.org/book/ordering-relation.html

```
<http://data.globalchange.gov/gcis.owl> a owl:Ontology
  dcterms:creator gcis:xiaogang_ma, gcis:curt_tilmes, gcis:peter_fox .  # <-- not ordered

[]  a gcis:QualifiedCreatorRelationship ;
  gcis:sequencePosition "1"^xsd ;  # <-- asserts position in ordered sequence
  gcis:creator gcis:xiaogang_ma ;
  gcis:created  <http://data.globalchange.gov/gcis.owl> .

[]  a gcis:QualifiedCreatorRelationship ;
  gcis:sequencePosition "2"^xsd ;
  gcis:creator gcis:curt_tilmes ;
  gcis:created  <http://data.globalchange.gov/gcis.owl> .

# ...
```

This modeling works well with SPARQL and can be used to determine the ordering of the members in the list.  I used the gcis namespace for the properties used in the qualified relationship resource because I do not know of a standard vocabulary to use in this case.  The modeling is common but most communities mint their own predicates to implement it.

**Comment by bduggan** [comment]
So far, for this particular case, we have found it more important to order groups of people by roles than to order individuals:

 http://data.globalchange.gov/report/nca3/chapter/our-changing-climate.thtml

(Note, it uses prov:qualifiedAttribution, and we have a finite set of "role"s.  )

But, there is no order specified in the representation -- could multiple people have the same gcis:sequencePosition?  If so, @justgo129 could add this field to the  [contributors template](https://github.com/USGCRP/gcis/blob/master/lib/Tuba/files/templates/contributors.ttl.tut).

**Comment by zednis** [comment]
Using PROV's qualifiedAttribution is probably the best way to express this qualified relation using an existing vocabulary.

On a side note, prov:hadRole is also an object-property and should not be used with a literal value.

This is what the modeling would look like using prov:qualifiedAttrbution.  I like it.

```
<http://data.globalchange.gov/gcis.owl> a owl:Ontology
  dcterms:creator gcis:xiaogang_ma, gcis:curt_tilmes, gcis:peter_fox ;
  prov:prov:qualifiedAttribution [
    a prov:Attribution ;
    prov:agent gcis:xiaogang_ma ;
    prov:hadRole gcis:creator ;
    gcis:sequencePosition "1"^^xsd:int
  ],
  [
    a prov:Attribution ;
    prov:agent gcis:curt_tilmes ;
    prov:hadRole gcis:creator ;
    gcis:sequencePosition "2"^^xsd:int
  ],
  [
    a prov:Attribution ;
    prov:agent gcis:peter_fox ;
    prov:hadRole gcis:creator ;
    gcis:sequencePosition "3"^^xsd:int
  ] .

gcis:creator
  a prov:Role ;
  rdfs:label "creator" .

# ...
```

**Comment by justgo129** [comment]
Thanks for catching the issue prov:hadRole.  Searching on sindice, specifically:
http://sindice.com/search?q=role

The sixth item that comes up is a TW term "has role."  Unfortunately, the linked to page is no longer available.  Is that also an object property?

**Comment by zednis** [comment]
@justgo129 I would suggest creating a new ticket for the prov:hadRole usage in the instance data, but I am not sure if this repository is the best place to create it.

As for the property, you can see more information on it at the [W3C PROV-O Recommendation](http://www.w3.org/TR/prov-o/#hadRole).

As I look at the definition of prov:hadRole I find it interesting that prov:Attribution is not in the property's domain.  There was a recent discussion on the prov-comments mailinglist about just this issue.  http://lists.w3.org/Archives/Public/public-prov-comments/2015Apr/0001.html

I think we should discuss this further on a thread dedicated to the usage of prov:hadRole.

**Comment by zednis** [comment]
Since there is not currently a clear requirement on ordering of creator and contributor individuals I am going to open a pull request (essentially a code review) on proposed changes to update dcterms contributor and creator to use foaf:Person instances.  This update will not include qualified relations with ordering information.

**Comment by bduggan** [comment]
On Wednesday, May  6, Stephan Zednik wrote: 

> @justgo129 I would suggest creating a new ticket for the prov:hadRole usage in the instance data, but I am not sure if this repository is the best place to create it.

You can make a ticket in the gcis repository; I will
add support for /role and /role/:role_identifier
endpoints.

Brian

**Comment by bduggan** [comment]
Would it also make sense to add a vivo:orcidId for contributors?  This would be consistent with [other](http://data.globalchange.gov/person/1025.thtml) instance data.

**Comment by justgo129** [comment]
I like that idea very much, Brian, for the reasons presented.

**Comment by justgo129** [comment]
Opened as an issue in the gcis repo, (Issue #141), see link above.

**Comment by zednis** [comment]
I closed pull request #3, which updated the usage of dcterms properties to use foaf:Person instances instead of literals.  This update resolves the core issue of this ticket.  I did not include a qualified attribution with sequence information because our discussion determined that this was not currently a requirement.  A good discussion, we can come back to it in a new ticket if need be.  With #3 closed I am closing this ticket.
